[WIP] Dynamic graphs

CDuPlooy opened the pull request:
On mobile.

#### Screenshot
![screen shot 2018-10-08 at 11 07 21](https://user-images.githubusercontent.com/5778739/46601171-a7234800-caec-11e8-88c4-af7a829a6040.png)

Commit:
Tasks graph is dynamic

Commit:
prettier

Commit:
Fixed test cases

Commit:
I suppose its time to head to bed.

Commit:
Count types && Count Nodes

Commit:
prettier

Commit:
Proper unsubscription on navigation.

Method used in #172

egeldenhuys commented on src/app/pages/services/service-detail-view/service-detail-view.component.ts:
```diff
@@ -87,15 +87,20 @@ export class ServiceDetailViewComponent implements OnInit {
-    this.image = this.serviceModel.Spec.TaskTemplate.ContainerSpec.Image.split('@')[0];
+    this.image = this.serviceModel.Spec.TaskTemplate.ContainerSpec.Image.split(
+      '@'
+    )[0];
```
Did `prettier` change this formatting?

egeldenhuys reviewed (changes requested):
The Offline/Online nodes chart is fine :+1: , however the chart on the left is comparing values that cannot be compared. The number of nodes compared to the number of networks does not provide any insights except "I have 1 node and 35 networks". Can we remove this chart or replace it with boxes showing the number of Nodes/networks/etc ?

The colour scheme is also better :+1:

CDuPlooy commented:
I have absolutely no idea. X)

As per the chart; I'll make those changes when I have time :)

annamarieHelberg commented:
I think taking a look at swarmpit and portainer's graphs would help. But thank you for implementing it :+1:

egeldenhuys committed:
Merge branch 'develop' into dynamic_graphs

egeldenhuys reviewed (changes requested):
I realise logic tends to leave us at such late times, but can you please give 8a232da2233425481d84eed7c5096ac69571a462 a relevant comment?

Commit:
Removed silly graph

Commit:
Rows

Commit:
prettier

Commit:
colours

Commit:
prettier

CDuPlooy commented:
> I think taking a look at swarmpit and portainer's graphs would help. But thank you for implementing it

I something like [this](https://github.com/swarmstack/swarmstack) would be nice.

I've removed the redundant graph and added cards.

![](https://i.imgur.com/6ohOxhk.png)

egeldenhuys committed:
Fix HTML formatting

This commit was actually to trigger a Docker Cloud build

Commit:
Revert tasks endpoint

FJMentz committed:
Minor changes to the graphs page

FJMentz committed:
Added another graph and made the cards route to the appropriate page

FJMentz committed:
prettier

CDuPlooy committed:
Changed interval

Due to the amount of requests being made I've increased the length of the interval.

egeldenhuys commented on src/app/pages/home/home.component.ts:
```diff
@@ -73,7 +73,7 @@ export class HomeComponent implements OnInit, OnDestroy {
     this.updateChartB();
     this.count_types();
-    this.page_start = interval(5000).subscribe(() => {
+    this.page_start = interval(30000).subscribe(() => {
```
This should help

CDuPlooy committed:
Merge branch 'develop' into dynamic_graphs

egeldenhuys reviewed (approved):
Looks good

egeldenhuys committed:
Merge branch 'develop' into dynamic_graphs
